org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/editor/contentassist/IdeContentProposalAcceptor.java

```diff
@@ -24,11 +26,25 @@
 public class IdeContentProposalAcceptor
 		implements IIdeContentProposalAcceptor, Comparator<Pair<Integer, ContentAssistEntry>> {
 
+	@Inject
+	private OperationCanceledManager operationCanceledManager;
+
+	protected CancelIndicator cancelIndicator;
+
+
 	protected final Set<Pair<Integer, ContentAssistEntry>> entries = new TreeSet<Pair<Integer, ContentAssistEntry>>(
 			this);
 
+	/**
+	 * @since 2.28
+	 */
+	public void setCancelIndicator(CancelIndicator cancelIndicator) {
+		this.cancelIndicator = cancelIndicator;
+	}
+
 	@Override
 	public void accept(ContentAssistEntry entry, int priority) {
+		operationCanceledManager.checkCanceled(cancelIndicator);
```

Comment:
does this need a null check

Reply:
It should not need it because @Inject does not have an optional flag, so operationCanceledManager can never be null. If the injector cannot create it, it will raise an exception, but never set it to null

Reply:
i mean the cancelindicator. i checked it, it works fine https://github.com/eclipse/xtext-core/blob/14b0026ed167e66a7a5f2e205160e03ba8ee447f/org.eclipse.xtext/src/org/eclipse/xtext/service/OperationCanceledManager.java#L86

Reply:
`java.lang.NullPointerException
	at org.eclipse.xtext.ide.editor.contentassist.IdeContentProposalAcceptor.accept(IdeContentProposalAcceptor.java:48)`
:(

Reply:
```
50841336 [pool-1-thread-4] ERROR t.ide.server.concurrent.ReadRequest  - Error during request: 
java.lang.NullPointerException
	at org.eclipse.xtext.ide.editor.contentassist.IdeContentProposalAcceptor.accept(IdeContentProposalAcceptor.java:48)
	at org.eclipse.xtext.ide.editor.contentassist.IdeContentProposalProvider._createProposals(IdeContentProposalProvider.java:123)
	at ai.five.assurance.dhc.ide.editor.contentassist.DhcContentProposalProvider.createProposals(DhcContentProposalProvider.java:283)
	at org.eclipse.xtext.ide.editor.contentassist.IdeContentProposalProvider.createProposals(IdeContentProposalProvider.java:71)
	at org.eclipse.xtext.ide.server.contentassist.ContentAssistService.createProposals(ContentAssistService.java:106)
	at org.eclipse.xtext.ide.server.contentassist.ContentAssistService.createCompletionList(ContentAssistService.java:79)
	at org.eclipse.xtext.ide.server.LanguageServerImpl.lambda$completion$23(LanguageServerImpl.java:567)
	at org.eclipse.xtext.ide.server.WorkspaceManager.doRead(WorkspaceManager.java:458)
	at org.eclipse.xtext.ide.server.LanguageServerImpl.completion(LanguageServerImpl.java:566)
	at org.eclipse.xtext.ide.server.LanguageServerImpl.lambda$completion$22(LanguageServerImpl.java:552)
	at org.eclipse.xtext.ide.server.concurrent.ReadRequest.lambda$doRun$0(ReadRequest.java:66)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
Sep 13, 2022 7:51:04 AM org.eclipse.lsp4j.jsonrpc.RemoteEndpoint fallbackResponseError
SEVERE: Internal error: java.lang.NullPointerException
java.util.concurrent.CompletionException: java.lang.NullPointerException
	at java.base/java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:331)
	at java.base/java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:346)
	at java.base/java.util.concurrent.CompletableFuture$UniAccept.tryFire(CompletableFuture.java:704)
	at java.base/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:506)
	at java.base/java.util.concurrent.CompletableFuture.completeExceptionally(CompletableFuture.java:2088)
	at org.eclipse.xtext.ide.server.concurrent.AbstractRequest.logAndCompleteExceptionally(AbstractRequest.java:73)
	at org.eclipse.xtext.ide.server.concurrent.ReadRequest.lambda$doRun$0(ReadRequest.java:69)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: java.lang.NullPointerException
	at org.eclipse.xtext.ide.editor.contentassist.IdeContentProposalAcceptor.accept(IdeContentProposalAcceptor.java:48)
	at org.eclipse.xtext.ide.editor.contentassist.IdeContentProposalProvider._createProposals(IdeContentProposalProvider.java:123)
	at ai.five.assurance.dhc.ide.editor.contentassist.DhcContentProposalProvider.createProposals(DhcContentProposalProvider.java:283)
	at org.eclipse.xtext.ide.editor.contentassist.IdeContentProposalProvider.createProposals(IdeContentProposalProvider.java:71)
	at org.eclipse.xtext.ide.server.contentassist.ContentAssistService.createProposals(ContentAssistService.java:106)
	at org.eclipse.xtext.ide.server.contentassist.ContentAssistService.createCompletionList(ContentAssistService.java:79)
	at org.eclipse.xtext.ide.server.LanguageServerImpl.lambda$completion$23(LanguageServerImpl.java:567)
	at org.eclipse.xtext.ide.server.WorkspaceManager.doRead(WorkspaceManager.java:458)
	at org.eclipse.xtext.ide.server.LanguageServerImpl.completion(LanguageServerImpl.java:566)
	at org.eclipse.xtext.ide.server.LanguageServerImpl.lambda$completion$22(LanguageServerImpl.java:552)
	at org.eclipse.xtext.ide.server.concurrent.ReadRequest.lambda$doRun$0(ReadRequest.java:66)
	... 5 more
```
The manager is null, but I agree with the initial suggestion that it absolutely should not be, or some injection problems should be visible.

Reply:
can you find out why it is null? how do you create / implement DhcContentProposalProvider

Reply:
I should be able to dig into this today and potentially open a PR (unless it is something unusual on our side of course). DhcContentProposalProvider is a fairly simple subclass, bound in the IdeModule for the language. Either way, I'll update here.